src/java.base/share/classes/java/lang/Process.java

```diff
@@ -475,6 +476,35 @@ public boolean waitFor(long timeout, TimeUnit unit)
         return false;
     }
 
+    /**
+     * Causes the current thread to wait, if necessary, until the
+     * process represented by this {@code Process} object has
+     * terminated, or the specified waiting duration elapses.
+     *
+     * <p>If the process has already terminated then this method returns
+     * immediately with the value {@code true}.  If the process has not
+     * terminated and the duration is not positive, then
+     * this method returns immediately with the value {@code false}.
+     *
+     * <p>The default implementation of this method polls the {@code exitValue}
+     * to check if the process has terminated. Concrete implementations of this
+     * class are strongly encouraged to override this method with a more
+     * efficient implementation.
+     *
+     * @param duration the maximum duration to wait; if not positive,
+     *                this method returns immediately.
+     * @return {@code true} if the process has exited and {@code false} if
+     *         the waiting duration elapsed before the process has exited.
+     * @throws InterruptedException if the current thread is interrupted
+     *         while waiting.
+     * @throws NullPointerException if duration is null
+     * @since 24
+     */
+    public boolean waitFor(Duration duration) throws InterruptedException {
+        Objects.requireNonNull(duration, "duration");
```

Comment:
Does this explicit null-check pull its weight here?

Reply:
Maybe not, but when this method gets a custom implementation, people don't have to remember to add a null check at front so there's no path that misses the NPE. It isn't wrong, so I don't see a problem with this.

Reply:
Maybe it's a personal preference, but I think it is a good practice to explicitly check non-null for public APIs. Some recent methods in that class, such as `inputReader(Charset)` do it, so there are precedences.